util/helpers.js

```diff
@@ -1,49 +1,139 @@
-const log4js  = require('log4js');
-const LOG     = log4js.getLogger('helpers');
-const format  = require('format');
-const _       = require('lodash');
+const log4js = require('log4js');
+
+const LOG = log4js.getLogger('helpers');
+const DEBUG = log4js.getLogger('debug');
+const _ = require('lodash');
+const emojiRegex = require('emoji-regex');
+
+const isDebug = process.env.MEME_USER;
 
 // Using getMessages will automatically use already cached messages when available,
 //     and will fetch more messages only if needed. This way we don't have to rely
 //     on the promise function structure every time we want to retrieve a message
 exports.getMessages = async function getMessages(message) {
-
   // Current messages in the Discord client's cache
   let cachedMessages = message.channel.messages;
-  
+
   // If there are only a couple of items in the cache...
-  if (cachedMessages.array().length < 10) {
-    LOG.info('Message cache empty for this channel; fetching messages.');
+  if (cachedMessages.array().length < 20) {
     // Use fetchMessages() to get messages from before we started caching
-    cachedMessages = await message.channel.fetchMessages({limit: 100});
+    cachedMessages = await message.channel.fetchMessages({ limit: 100 });
   }
 
   return cachedMessages;
-}
+};
+
 
+// Randomly look through all results until we get a message that
+//     isnt bot generated, isn't a bot command, and the message content isn't blank
+exports.randomUserMessage = function filterMessage(messages, options = {}) {
+  exports.debug('Getting random user message');
 
-// Randomly look through all results until we get a message that 
-//     isnt bot generated or a bot command
-exports.randomUserMessage = function filterMessage(messages) {
   // Shuffle messages first
-  let newMessages = _.shuffle(messages.array());
-  
+  const newMessages = _.shuffle(messages.array());
   // Default to first message
-  let found = newMessages[0];
-  
-  // Skip loop if first message matches. Otherwise, go through all messages
-  //     until we find a suitable math
-  let i = 1;
-  while ( (found.author.bot || found.content.startsWith('.')) && i < newMessages.length) {
+  let found = null;
+
+  // Go through all messages until we find a suitable math
+  let i = 0;
+  let isNatural = true;
+
+  while (!found || ((found.author.bot || found.content.startsWith('.') || found.content.trim() === '' || !isNatural) && i < newMessages.length)) {
```

Comment:
Does this need to be a while loop since you're already iterating through i? Not seeing any immediate cases where this could loop indefinitely, but worth thinking about.

Reply:
Yes; the loop only continues if a match is found, so it only iterates to the next record if the current message doesn't fit the criteria. The loop is set self-cancel once it reaches the end of the available messages.